csv_detective/detect_fields/temp/date/__init__.py

```diff
@@ -1,46 +1,30 @@
-import re
-from dateutil.parser import parse, ParserError
-from csv_detective.detect_fields.other.float import _is as is_float
-from unidecode import unidecode
+from datetime import datetime
+from typing import Optional
+
+from dateparser import parse as date_parser
+from dateutil.parser import parse as dateutil_parser, ParserError
 
 PROPORTION = 1
 # /!\ this is only for dates, not datetimes which are handled by other utils
 
 
-def is_dateutil_date(val: str) -> bool:
-    # we don't want to get datetimes here, so length restriction
-    # longest date string expected here is DD-septembre-YYYY, so 17 characters
-    if len(val) > 17:
-        return False
+def date_casting(val: str) -> Optional[datetime]:
+    """For performance reasons, we try first with dateutil and fallback on dateparser"""
     try:
-        res = parse(val, fuzzy=False)
```

Comment:
Way cleaner! Is that dateparser/dateutil.parse fallback idea taken from hydra codebase?

Reply:
Yes, I got this from [here](https://github.com/datagouv/hydra/blob/56f5e1209d0932b34407c26df088a826ce5026d4/udata_hydra/analysis/helpers.py#L13), I thought it looked much cleaner too, and the tests pass perfectly well with this refactor ✅